cmd/launcher/uninstall_darwin.go

```diff
@@ -0,0 +1,47 @@
+//go:build darwin
+// +build darwin
+
+package main
+
+import (
+	"context"
+	"fmt"
+	"os"
+	"os/exec"
+)
+
+func removeLauncher(ctx context.Context, identifier string) error {
+	if identifier == "" {
+		// Ensure identifier is non-empty and use the default if none provided
+		identifier = "kolide-k2"
+	}
+
+	launchDaemonPList := fmt.Sprintf("/Library/LaunchDaemons/com.%s.launcher.plist", identifier)
+	launchCtlPath := "/bin/launchctl"
+	launchCtlArgs := []string{"unload", launchDaemonPList}
+
+	cmd := exec.CommandContext(ctx, launchCtlPath, launchCtlArgs...)
+	if err := cmd.Run(); err != nil {
+		return err
```

Comment:
This is fine, but I think it probably won't expose whatever error message `launchctl` spits out

Reply:
Good point, I've added code to log `launchctl`/`systemctl` output.